kqueen/auth/ldap.py

```diff
@@ -61,30 +92,42 @@ def verify(self, user, password):
             None: Invalid credentials, Authentication failure.
             User: authenticated user.
         """
+        matched_dn = self._get_matched_dn(user.username)
+
+        for dn in matched_dn:
+            if self._bind(dn, password):
+                full_dn = dn
+
+        if full_dn:
+            logger.debug('Valid full-DN found: {}. It will be stored in user metadata'.format(full_dn))
+            return user, None
+
+        return None, "Verification failed"
 
-        dn = self._email_to_dn(user.username)
+    def _bind(self, dn, password):
 
         try:
+            self.connection = ldap.initialize(self.uri)
             bind = self.connection.simple_bind_s(dn, password)
 
             if bind:
-                return user, None
+                return True
```

Comment:
Initially this method return username and error message, you reworked it and now it is returned two or one value, which is inappropriate

Reply:
oh, thanks, i need to switch all return to 1-value arch, yes?

Reply:
done